doc/source/data/glossary.rst

```diff
@@ -0,0 +1,140 @@
+        A processing unit of data. A :class:`~ray.data.Dataset` consists of a list of
+        blocks.
+
+        Under the hood, :term:`Datasets <Datasets (library)>` partition :term:`records <Record>`
+        into a set of distributed data blocks. This allows Datasets to perform operations
+        in parallel.
+
+        Unlike a batch, which is a user-facing object, a block is an internal abstraction.
+
+    Block format
+        The way :term:`blocks <Block>` are represented.
+
+        Blocks are represented as Arrow tables, pandas DataFrames, and Python lists.
+        To determine the block format, call :meth:`~ray.data.Dataset.dataset_format`.
+
+    Data Shuffling
+        The act of reordering :term:`records <Record>`. Shuffling prevents overfitting of models.
+
+        To perform a global shuffle, call :meth:`~ray.data.Dataset.random_shuffle()`.
+
+        .. doctest::
+
+            >>> dataset = ray.data.range(10)
+            >>> dataset.random_shuffle().take_all()
+            [4, 7, 1, 5, 2, 3, 8, 6, 9, 0]
+
+        If :meth:`~ray.data.Dataset.random_shuffle()` is slow, perform a local shuffle.
+        It’s lower quality but more performant. To perform a local shuffle, call methods
+        like :meth:`~ray.data.Dataset.iter_batches` and specify ``local_shuffle_buffer_size``.
+
+        .. doctest::
+
+            >>> dataset = ray.data.range(10)
+            >>> next(iter(dataset.iter_batches(local_shuffle_buffer_size=5)))
+            [2, 6, 7, 8, 4, 9, 3, 1, 5, 0]
+
+        To learn more about shuffling data, read :ref:`Shuffling Data <air-shuffle>`.
+
+    Datasets (library)
+        A library for distributed data processing.
+
+        Datasets isn’t intended as a replacement for more general data processing systems.
+        Its utility is as the last-mile bridge from ETL pipeline outputs to distributed applications and libraries in Ray.
+
+        To learn more about Ray Datasets, read :ref:`Key Concepts <dataset_concept>`.
+
+    Dataset (object)
+        A class that represents a distributed collection of data.
+
+        :class:`Datasets <ray.data.Dataset>` expose methods to read, transform, and consume data at scale.
+
+        To learn more about Datasets and the operations they support, read the :ref:`Datasets API Reference <data-api>`.
+
+    Datasource
+        A :class:`~ray.data.Datasource` specifies how to read and write from
+        a variety of external storage and data formats.
+        For example: Parquet, images, TFRecord, CSV, and MongoDB.
+
+        To learn more about Datasources, read :ref:`Creating a Custom Datasource <custom_datasources>`.
+
+    Record
+        A single data item.
+
+        If your dataset is :term:`tabular <Tabular Dataset>`, then records are :class:`TableRows <ray.data.row.TableRow>`.
+        If your dataset is :term:`simple <Simple Dataset>`, then records are arbitrary Python objects.
+        If your dataset is :term:`tensor <Tensor Dataset>`, then records are ndarrays.
```

Comment:
nit: ndarrays -> NumPy ndarrays.

Also add a link to https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html .